include/glaze/core/wrappers.hpp

```diff
@@ -71,14 +71,16 @@ namespace glz
          };
       }
 
+      // Used for determining the default value
       struct deduct_default_t
       {};
 
       template <auto MemPtr, auto Default, class T>
       bool is_default(const T& val)
       {
          if constexpr (std::same_as<std::decay_t<decltype(Default)>, deduct_default_t>) {
-            return val.*MemPtr == T{}.*MemPtr;
+            static thread_local auto defaulted = T{};
```

Comment:
I believe the `thread_local` is unnecessary because C++ requires static variable initialization to be thread safe since C++11. The statement might be better as follows.
```
static const auto defaulted = T{}.*MemPtr;
return val.*MemPtr == defaulted;
```
I expected the compiler can be smart enough not to construct unused data members this way; however, based on my experiment on compiler explorer, no compiler can do that.

Reply:
You are correct, because we aren't changing state, we can have a single, global static variable. I'm just used to requiring `thread_local` for mutation. We can also make it `const`.